tests/config/test_defaults.py

```diff
@@ -41,93 +42,194 @@ def test_create_defaults_without_permission() -> None:
-    ("section", "item", "delimiter", "strip", "duplicated_ok", "expect"),
+    ("user_config_input", "delimiter", "strip", "expect"),
     [
         (
-            "test.list",
-            "commas_string",
+            """
+            [test.list]
+            list = ,github.com, gitlab.com, space string, space string
+            """,
             ",",
             False,
+            ["", "github.com", " gitlab.com", " space string", " space string"],
+        ),
+        (
+            """
+            [test.list]
+            list = ,github.com, gitlab.com, space string, space string
+            """,
+            ",",
             True,
-            ["", " gitlab.com", " space string", " space string", "github.com"],
+            ["github.com", "gitlab.com", "space string", "space string"],
         ),
-        ("test.list", "commas_string", ",", False, False, ["", " gitlab.com", " space string", "github.com"]),
-        ("test.list", "commas_string", ",", True, True, ["github.com", "gitlab.com", "space string", "space string"]),
-        ("test.list", "commas_string", ",", True, False, ["github.com", "gitlab.com", "space string"]),
-        # Using None as the delimiter parameter will ignore cleanup
-        ("test.list", "default", None, True, False, ["comma_ended,", "github.com", "space", "string"]),
-        ("test.list", "default", None, False, False, ["comma_ended,", "github.com", "space", "string"]),
+        # Using None as the `delimiter` will also remove leading and trailing spaces of elements. Therefore,
+        # the results must be the same whether `strip` is set to True or False.
         (
-            "test.list",
-            "default",
+            """
+            [test.list]
+            list =
+                github.com
+                comma_ended,
+                space string
+                space string
+            """,
             None,
-            False,
             True,
-            ["comma_ended,", "github.com", "space", "space", "string", "string"],
+            ["github.com", "comma_ended,", "space", "string", "space", "string"],
         ),
-        ("test.list", "one_line", None, True, False, ["comma_ended,", "github.com", "space", "string"]),
         (
-            "test.list",
-            "one_line",
+            """
+            [test.list]
+            list =
+                github.com
+                comma_ended,
+                space string
+                space string
+            """,
+            None,
+            False,
+            ["github.com", "comma_ended,", "space", "string", "space", "string"],
+        ),
+        (
+            """
+            [test.list]
+            list = github.com comma_ended, space string space string
+            """,
             None,
             True,
-            True,
-            ["comma_ended,", "github.com", "space", "space", "string", "string"],
+            ["github.com", "comma_ended,", "space", "string", "space", "string"],
```

Comment:
I feel like this third set of inputs could be removed as it doesn't really test anything new. That multiple list items can be read per line is already handled by the above two inputs.

Reply:
The reason for this test case was for when `None` is used as the delimiter. `None` as the delimiter used in `str.split` will split the string on any white spaces, new lines, etc ([docs](https://docs.python.org/3/library/stdtypes.html#str.split)). This test case is for validating if the string is splits correctly on spaces. 
I agree that it doesn't test anything new in the context of `defaults.get_list` and therefore it can be removed.

Reply:
Removed in https://github.com/oracle/macaron/pull/660/commits/8a3f5ce40e5a0e8288b2205ff445d075aaa8830a